### PR TITLE
Content negotiation fails in subsequent requests accepting same media type

### PR DESCRIPTION
**Pull request by dougbu**
- #3140
- clone `MediaTypeHeaderValue` instance before updating it when content negotiation succeeds
  - avoids changes to `MediaTypeConstants` properties and `OutputFormatter.SupportedMediaTypes` entries
  - `MediaTypeHeaderValue.Clone()` does not exist in our DNX Core fork of this class
  - in previous implementation, was called defensively rather than when required
- update `WebApiCompatShimBasicTest` functional tests to use `MvcTestFixture<TStartup>` everywhere
  - #3140 blocked that final migration
  - remove `TestHelper` since it's no longer referenced

nits:
- remove comments mentioning `TestHelper`
- correct spelling of "negotiation"

### CODE REVIEW

**Comment by dougbu** [comment]
# remove `TestHelper` since it's no longer referenced

Just in case you missed it.

File: src/Microsoft.AspNet.Mvc.WebApiCompatShim/ContentNegotiator/DefaultContentNegotiator.cs [hunk 1]
```diff
@@ -85,14 +83,25 @@ public DefaultContentNegotiator(bool excludeMatchOnTypeOnly)
...
                 {
-                    bestFormatterMatch.MediaType.CharSet = bestEncodingMatch.WebName;
+                    // Clone media type value since this is not done defensively in this implementation.
+                    // `MediaTypeHeaderValue` lacks a Clone() method in this runtime. Fortunately, this is the only
+                    // update to an existing instance we need.
+                    var clonedMediaType = new MediaTypeHeaderValue(bestMediaType.MediaType);
+                    foreach (var parameter in bestMediaType.Parameters)
+                    {
+                        clonedMediaType.Parameters.Add(new NameValueHeaderValue(parameter.Name, parameter.Value));
+                    }
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.WebApiCompatShim/ContentNegotiator/DefaultContentNegotiator.cs]: Annoying that the the copy constructor is protected here.

**Comment by rynowak** [comment]
:shipit:

**Comment by dougbu** [comment]
53060be

**Pull request closed without merging**